doc/ins_sign_catalyst_registration.md

```diff
@@ -0,0 +1,89 @@
+# Catalyst Voting Key Registration
+
+## Description
+
+Cardano uses a sidechain for its treasury system. One needs to "register" to participate on this sidechain by submitting a registration transaction on the mainnet chain. This CIP details the registration transaction format.
+
+A Catalyst registration transaction are a regular Cardano transaction with a specific transaction metadata associated with it.
+
+For more details about Catalyst registration see [CIP-0015](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0015/CIP-0015.md)
+
+---
+
+In the following list of APDU messages (which are to be sent in the listed order), we only give the value of P2 and the format of message data. The general format of the message is
+
+|Field|Value|
+|-----|-----|
+| CLA | `0xD7` |
+| INS | `0x21` |
+|  P1 | `0x08` |
+|  P2 | (specific for each subcall) |
+
+All the resposes are empty.
+
+---
+
+**Voting key**
+
+P2 = `0x30`
+
+*Data*
+
+|Field| Length | Comments|
+|-----|--------|---------|
+|Voting public key |  32 | |
+
+---
+
+**Staking key**
+
+P2 = `0x31`
+
+*Data*
+
+|Field| Length | Comments|
+|-----|--------|---------|
+|staking key path      | variable | BIP44 path. See [GetExtPubKey call](ins_get_extended_public_key.md) for a format example |
+
+---
+
+**Voting rewards address**
+
+P2 = `0x32`
+
+*Data*
+
+|Field| Length | Comments|
+|-----|--------|--------|
+|Address params | variable | see `view_parseAddressParams` in [src/addressUtilsShelley.c](../src/addressUtilsShelley.c)|
+
+Note: Only Shelley-era address parameters are accepted.
+
+---
+
+**Nonce**
+
+P2 = `0x33`
+
+*Data*
+
+|Field| Length | Comments|
+|-----|--------|--------|
+|Nonce| 8| Big endian|
```

Comment:
Hm. Don't we specify if it is int or uint? Apparently not, but I guess we should, everywhere. (Int is rare, will be used in for multiasset in minting. I guess everything until now was uint.) Perhaps create an issue to fix it later?

Reply:
fair point, I made an issue for it: https://github.com/vacuumlabs/ledger-app-cardano-shelley/issues/69